### ISSUE TITLE
Consul constraints don't use non-default Consul cluster (ENT)

### ISSUE THREAD

**Issue by SuyashHashiCorp** [hashicorp/nomad#19442]
I have encountered an issue where Multi-Cluster Consul is not working as expected when using a custom name in conjunction with Nomad version 1.7.0-rc.1+ent.

### Nomad version
Output from `nomad version`
`v1.7.0-rc.1+ent`

### Consul version
Output from `consul version`
`v1.17.0+ent`

### Operating system and Environment details
OS - Linux (Ubuntu) - `amd64` and `arm64` both

### Reproduction steps
- Create two Consul clusters with version v1.17.0+ent.

- Provision Nomad cluster (1 server and 1 client) with version v1.7.0-rc.1+ent. 

- Use the below consul blocks configuration for Nomad Server and Client Nodes -

```
consul {
  address = "10.211.55.11:8500"
  name = "alpha"
}

consul {
  address = "10.211.55.12:8500"
  name = "beta"
}
```

- Start the Nomad services on the Nomad server and client nodes.

- Not able to see the services in the Consul UI Dashboard or consul catalog services in both the consul clusters. 

- In the Nlmad node logs getting the below warning - 

```
Dec 05 12:37:38 c1 nomad[4568]:     2023-12-05T12:37:38.777+0530 [WARN]  client.fingerprint_mgr.consul: malformed Consul response prevented linking
Dec 05 12:37:38 c1 nomad[4568]:     2023-12-05T12:37:38.787+0530 [WARN]  client.fingerprint_mgr.consul: malformed Consul response prevented linking
```

- While checking on CLI using the Nomad command nomad node status -verbose <node_id> | grep -i consul I can see the details of Consul integration with Nomad. Here is the sample snippet for your reference- 

```
$ nomad server members
Name       Address      Port  Status  Leader  Raft Version  Build           Datacenter  Region
s1.global  10.211.55.6  4648  alive   true    3             1.7.0-rc.1+ent  dc1         global
```

```
$ nomad node status
ID        Node Pool  DC   Name  Class   Drain  Eligibility  Status
2f97f440  default    dc1  c1    <none>  false  eligible     ready
```

```
$ nomad node status -verbose 2f97f440 | grep -i consul
consul.alpha.connect             = true
consul.alpha.datacenter          = dc1
consul.alpha.ft.namespaces       = true
consul.alpha.grpc                = 8502
consul.alpha.revision            = 9afb1901
consul.alpha.server              = false
consul.alpha.sku                 = ent
consul.alpha.version             = 1.17.0+ent
consul.beta.connect              = true
consul.beta.datacenter           = dc1
consul.beta.ft.namespaces        = true
consul.beta.grpc                 = 8502
consul.beta.revision             = 9afb1901
consul.beta.server               = false
consul.beta.sku                  = ent
consul.beta.version              = 1.17.0+ent
```

- While deploying the job I got the below error - 

```
$ nomad job plan example.nomad.hcl
+ Job: "example"
+ Task Group: "cache" (1 create)
  + Task: "redis" (forces create)

Scheduler dry-run:
- WARNING: Failed to place all allocations.
  Task Group "cache" (failed to place 1 allocation):
    * Constraint "${attr.consul.version} semver >= 1.8.0": 1 nodes excluded by filter

Job Modify Index: 0
To submit the job with version verification run:

nomad job run -check-index 0 example.nomad.hcl

When running the job with the check-index flag, the job will only be run if the
job modify index given matches the server-side version. If the index has
changed, another user has modified the job and the plan's results are
potentially invalid.
```

#### Expected Result
- Noamd job should be running properly.
- The Nomad server and client need to be registered in the Consul dashboard and catalog services.

#### Actual Result
- Consul UI or CLI is not showing the services.
- Fingerprint is getting registered in Consul, however, it is not getting populated in the Consul catalog.
- Even Nomad client nodes are getting consul integration as shown in the above reproduction steps, however, Nomad jobs are getting `* Constraint "${attr.consul.version} semver >= 1.8.0": 1 nodes excluded by filter` error.

### Job file (if appropriate)
```
job "example" {
  group "cache" {
    consul {
      cluster = "beta"
    }

    network {
      port "db" {
        to = 6379
      }
    }

    service {
      name = "redis"
      port = "db"
    }

    task "redis" {
      driver = "docker"

      config {
        image          = "redis:7"
        ports          = ["db"]
        auth_soft_fail = true
      }

      identity {
        env  = true
        file = true
      }

      resources {
        cpu    = 500
        memory = 256
      }
    }
  }
}
```

Thanks..!!

**Comment by tgross** [comment]
Hi @SuyashHashiCorp! If you hit the [Agent Self API](https://developer.hashicorp.com/nomad/api-docs/agent#query-self) endpoint for your node, you'll see that there's a default Consul block getting configured because you left one out of your own configuration. We failed to document this (and we should fix it), but the expectation that if Consul is in use there's always going to be a `consul` block for `name = "default"` (or unset).

Because the `default` cluster is set to the default Consul API values, and because you probably don't have a Consul cluster listening on that address, it can't fingerprint the `default` cluster and that's why you only see the `alpha` and `beta` cluster in the node attributes.

However, there's still a bug here! Which is that when we set the constraint for the allocation, we should be setting it based on the expected attribute for the cluster needed by the task group, not for the default cluster. I'll update the title of this issue to reflect that and get this fixed.
